upgrade_test.py

```diff
@@ -694,44 +698,48 @@ def test_generic_cluster_upgrade(self):  # pylint: disable=too-many-locals,too-m
 
         upgrade_sstables = self.params.get('upgrade_sstables')
 
-        if num_nodes_to_rollback > 0:
-            # Upgrade all nodes that should be rollback later
-            for i in range(num_nodes_to_rollback):
-                InfoEvent(f'Step{i + 1} - Upgrade node{i + 1}')
-                self.db_cluster.node_to_upgrade = self.db_cluster.nodes[indexes[i]]
-                self.log.info(f'Upgrade Node {self.db_cluster.node_to_upgrade.name} begins')
-                self.upgrade_node(self.db_cluster.node_to_upgrade, upgrade_sstables=upgrade_sstables)
-                self.log.info(f'Upgrade Node {self.db_cluster.node_to_upgrade.name} ended')
+        def change_node_version(node_indexes, is_upgrade=True):
+            nonlocal step_idx
```

Comment:
why using nonlocal ? I see no real reason here

Reply:
I change the value of the `step_idx` variable that is not in my scope.

Reply:
than move it to a property on the test instance, this isn't a very clear way of do such a thing.
also the idea of step is kind of lost here, since now understand which step is which from the code, is very very hard like that.

the whole point of printing step numbers, is to make it easier to find it in the code.

Reply:
OK.
Should I delete it?

Reply:
The idea is to keep the code simple for others to read and extend, not making life harder by trying to understand why one did something like this.

Reply:
I revert my changes